src/pages/Product/index.js

```diff
@@ -1,9 +1,168 @@
+import {
+  changeQtyProduct,
+  enterQtyProductInDetails,
+  addToCart
+} from "../../action/cart";
+import Arrow from "../../icons/Arrow";
+import {
+  CHANGE_QTY_PRODUCT_DOWN,
+  CHANGE_QTY_PRODUCT_UP
+} from "../../constants";
+import Loading from "../../components/Loading";
+import ThreeArrows from "../../icons/ThreeArrows";
 
 class Product extends Component {
+  componentDidMount() {
+    const { history, getProduct, enterQtyProductInDetails } = this.props;
+    const { pathname } = history.location;
+    const idProduct = pathname.split("/")[2];
+    getProduct(idProduct);
+    enterQtyProductInDetails("");
+  }
+
   render() {
-    return <div />;
+    const {
+      productInfo,
+      cartInfo,
+      changeQtyProduct,
+      enterQtyProductInDetails,
+      chooseImg,
+      addToCart,
+      history
+    } = this.props;
+    const { pathname } = history.location;
+    const idProduct = pathname.split("/")[2];
+    const { loader, productData, sliderData } = productInfo;
+    const { qtyProduct } = cartInfo;
+    const { activeImg, imgList } = sliderData;
+    const { price, title, shortDescription, type } = productData;
+    return (
+      <div className={styles.wrapper}>
+        <div className={styles.nav}>
+          <h4 className={styles.linkTo}> shop</h4>
+          <ThreeArrows />
+          <h4 className={styles.linkTo}> {type} </h4>
+          <ThreeArrows />
+          <h4 className={styles.titleLink}> {title}</h4>
+        </div>
+        <Slider
+          listImg={imgList}
+          activeImg={activeImg}
+          chooseImg={chooseImg}
+          loader={loader}
+        />
+        <div className={styles.info}>
+          <div className={styles.price}>${price}</div>
+          <div className={styles.title}>{title}</div>
+          <div className={styles.description}>{shortDescription}</div>
+          <div className={styles.qty}>
+            <p className={styles.qtyTitle}>Qty</p>
+            <div className={styles.qtyCount}>
+              <input
+                className={styles.input}
+                onChange={event =>
+                  enterQtyProductInDetails(parseInt(event.target.value))
+                }
+                type="text"
+                placeholder="Input field"
+                value={qtyProduct}
+              />
+              <div className={styles.qtyControl}>
+                <div
+                  onClick={() => changeQtyProduct(CHANGE_QTY_PRODUCT_UP)}
+                  className={styles.qtyControlTop}
+                >
+                  <Arrow />
+                </div>
+                <div
+                  onClick={() => changeQtyProduct(CHANGE_QTY_PRODUCT_DOWN)}
+                  className={styles.qtyControlBottom}
+                >
+                  <Arrow />
+                </div>
+              </div>
+            </div>
+          </div>
+          <div
+            className={styles.btn}
+            onClick={() => btnStatus(idProduct, qtyProduct, addToCart)}
```

Comment:
btnStatus function is useless

`onClick={() => qtyProduct.length > 0 && addToCart(idProduct)}`

but if you use not div but button 
you can use component with props `disabled`
`<button disabled={ qtyProduct.length <= 0}>`